src/main/java/org/toradocu/translator/ConditionTranslator.java

```diff
@@ -630,6 +630,64 @@ private static String translateSecondPart(String text, DocumentedMethod method)
       translatePropositions(propositions, method);
       conditions.add(propositions.getTranslation());
     }
+
+    //If conditions are empty
+    boolean noSubject = false;
+
+    //Check if it is due to the subject
+    for (PropositionSeries p : extractedPropositions) {
+      noSubject = p.isEmpty();
+      for (Proposition prop : p.getPropositions()) {
+        if (prop.getSubject().getSubject().length() == 0) noSubject = true;
+      }
+    }
+
+    if (noSubject) {
+      //If it is, then add the subject as needed
+      //Check for the verb.
+      String[] possibleVerbs = {"is", "is not", "isn't", "are", "are not", "aren't"};
+      boolean containsVerb = false;
+      for (String possibleVerb : possibleVerbs) {
+        if (containsVerb = text.contains(" " + possibleVerb + " ")) {
+          //If no arguments, subject this
+          int numParam = method.getParameters().size();
+          if (numParam == 0) {
+            text.replace(possibleVerb, " this " + possibleVerb);
+          } else if (numParam == 1) {
+            text.replace(possibleVerb, "parameter " + possibleVerb);
+          } else if (numParam == 2) {
+            text.replace(
+                possibleVerb,
+                method.getParameters().get(0).getName()
+                    + " and "
+                    + method.getParameters().get(1).getName()
+                    + possibleVerb);
+          }
+        }
+      }
+
+      int numParam = method.getParameters().size();
+      if (numParam == 0) {
+        text = "this is" + text;
+      } else if (numParam == 1) {
+        text = method.getParameters().get(0).getName() + " is" + text;
+      } else if (numParam == 2) {
+        text =
+            method.getParameters().get(0).getName()
+                + " and "
+                + method.getParameters().get(1).getName()
+                + " are"
+                + text;
+      }
+    }
+    extractedPropositions = getPropositionSeries(text, method);
+    conditions = new LinkedHashSet<>();
+    // Identify Java code elements in propositions.
+    for (PropositionSeries propositions : extractedPropositions) {
+      translatePropositions(propositions, method);
```

Comment:
I'd keep only one invocation of `translatePropositions` at the end of the subject addition.